src/components/Proposal/Voting/VoteButtons.tsx

```diff
@@ -113,13 +113,15 @@ class VoteButtons extends React.Component<IProps, IState> {
         "Can't change your vote" :
         (currentAccountState && currentAccountState.reputation.eq(new BN(0))) ?
           "Requires reputation in this DAO" :
-          proposal.stage === IProposalStage.ExpiredInQueue ||
-              (proposal.stage === IProposalStage.Boosted && expired) ||
-              (proposal.stage === IProposalStage.QuietEndingPeriod && expired) ||
-              (proposal.stage === IProposalStage.Queued && expired) ?
-            "Can't vote on expired proposals" :
-            proposal.stage === IProposalStage.Executed ?
-              `Can't vote on ${proposal.winningOutcome === IProposalOutcome.Pass ? "passed" : "failed"} proposals` : "";
+          (currentAccountState && (proposal.createdAt < currentAccountState.createdAt)) ?
```

Comment:
This check doesn't guarantee that the account had reputation on proposal.createdAt.

Neither does `(currentAccountState && currentAccountState.reputation.eq(new BN(0)))` which seems like completely incorrect logic that needs to be corrected (it is checking the current balance but should be checking the balance at proposal creation time).

Am I missing something?

Reply:
- if this is true `(currentAccountState && currentAccountState.reputation.eq(new BN(0)))` so `"Requires reputation in this DAO"` should be the error msg. 
-otherwise if this is true `(currentAccountState && (proposal.createdAt < currentAccountState.createdAt)) ` so 
`"Must have had reputation in this DAO when the proposal was created"` should be the error msg.
whats wrong with that ?

Reply:
If `currentAccountState` can only ever be defined when the current account currently has rep in the DAO, then why are we checking `currentAccountState.reputation.eq(new BN(0)))`?  That case should never be able to be true.

Reply:
BTW, we should also be modifying the logic on line 98:

`const votingDisabled = [....]`

Also line 62:

`if (currentAccountState.reputation.gt(new BN(0)))`

Reply:
because if the current account is not a member of the dao a default currentAccountState value is returned from arc.js . [see here](https://github.com/daostack/arc.js/blob/51ce56ad8691dab37f0dfc7a13a9967cdfe7ac5b/src/member.ts#L184)

Reply:
Looking at the subgraph, it appears that `createdAt` represents the date of the last block in which the account's rep was minted or burned (pretty misleading if you ask me).

(see [here](https://github.com/daostack/subgraph/blob/c34fdbbdebea46a95a42c84609ec2b0900d425ed/src/mappings/Reputation/mapping.ts#L84)).

So `proposal.createdAt < currentAccountState.createdAt` is true when the proposal was created prior to the last time the amount of the account's reputation changed.  This tells us nothing about what the balance was at the time the proposal was created.

Reply:
yes, but it is set only at the first time see https://github.com/daostack/subgraph/blob/c34fdbbdebea46a95a42c84609ec2b0900d425ed/src/mappings/Reputation/mapping.ts#L42

Reply:
There remains an issue:   Suppose:

1) DAO and Reputation contract are created on date 1
2) User account is minted rep on date 2
3) User rep is burned (rage quit, penalty by vote of the DAO, or really poor voting patterns) to zero on date 3
4) Proposal is created on date 4
5) User rep is minted again to > zero on date 5
6) User tries to vote on date 6

The line in question will return false even though the user had zero rep on the date the proposal was created.

Reply:
if user rep is burned the rep member entity is deleted and the user is kiked out of the dao.
when it will be minted rep again the createDate will be upon this event.